lab-cortney/server.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const http = require('http');
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const parseBody = require('./lib/parseJson');
+
+const cowsay = require('cowsay');
+
+const port = process.argv[2] || 3000;
+
+
+const server = http.createServer(function(req, res){
+  req.url = parseUrl(req.url);
+  req.url.query = parseQuery(req.url.query);
+
+  if (req.url.pathname === '/' && !/(PUT|POST|DELETE)/.test(req.method)) {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write(cowsay.say({f: 'moofasa', text: 'Welcome, and moorawr! Let\'s type a query! Like this:'+
+    '\n'+'localhost:3000/cowsay?text=yeahletsgo'}));
+    res.end();
+    return;
+  }
+
+  if (req.url.pathname === '/cowsay?text=', req.url.query.text) {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write(cowsay.say({text: req.url.query.text}));
+    res.end();
+    return;
+  }
+
+  if(/(PUT|POST|DELETE)/.test(req.method)) {
```

Comment:
I did some texting on this portion of your code and it looks like you are not accessing the JSON object correctly.   Currently, if your request was "localhost:3000/cowsay"  and I send a POST request with a JSON object, the server will break. Take a close look at what you are trying to pass cowsay, that you're fetching JSON data properly.